h/api/test/auth_test.py

```diff
@@ -28,3 +31,150 @@ def test_translate_annotation_principals(p_in, p_out):
     result = auth.translate_annotation_principals(p_in)
 
     assert set(result) == set(p_out)
+
+
+generate_bearer_token_fixtures = pytest.mark.usefixtures('jwt')
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_calls_encode(jwt):
+    """It should pass the right arguments to encode()."""
+    before = datetime.datetime.utcnow()
+    request = mock_request()
+
+    auth.generate_bearer_token(request, 3600)
+
+    assert jwt.encode.call_args[0][0]['sub'] == 'acct:testuser@hypothes.is', (
+        "It should encode the userid as 'sub'")
+    after = datetime.datetime.utcnow() + datetime.timedelta(seconds=3600)
```

Comment:
It's not about the exact implementation of `ApproximateMatcher` -- you could easily implement it to do exactly the same assertion you're doing in your version.

It's about the difference between "assert it made this call" and `assert ...call_args[0][0].get('sub') == 'foo'`, which doesn't map easily onto how you're thinking about that call when writing the code under test.